flybywiresim/installer
# feat: check and warning for installed incompatible add-ons

## Summary of Changes
This PR adds a configurable list of incompatible add-ons to each installable add-on. 
If searches the configured community folder and reads each folder/manifest.json to compare to the incompatibility list. 
Incompatible add-ons are listed in a warning with the recommendation to remove them. 

- [x] Split configuration for Installation folder and Community folder (two separate configs now)
- [x] Incompatibility configuration structure
- [x] Implement search logic
- [x] Warning
- [x] CleanUp / Improve code

## Screenshots (if necessary)
Example data only! List of incompatible add-ons is defined in installer-data repo. 
![image](https://user-images.githubusercontent.com/16833201/190905703-581a2d2a-c518-403e-9731-31b09c2d21e7.png)

## Testing Instructions
- Make sure to use this Installer config: https://cdn.flybywiresim.com/installer/config/pr-6.json (ask on Discord how to change)
- Configuration of Install folder and Community folder has been separated - test this as best possible
- See list of current known incompatible add-ons - install them an test Installer warning (sufficient to create a folder and place the add-on's manifest.json file in there).

Discord username (if different from GitHub): Cdr_Maverick#6475
